### PR TITLE
let stateless-nonce use SHA-2

### PR DESCRIPTION
**Pull request by thomas-fossati**

### CODE REVIEW

File: draft-birkholz-rats-epoch-markers.md [hunk 1]
```diff
@@ -229,11 +235,11 @@ Timestamp:
...
 
 Pad:
-: pad bytes used to make the stateless nonce the desired size.
+: zero or more pad bytes, used to make the stateless nonce the desired size.
 
 AuthTag:
-: HMAC w/ SHA-1 computed over the CBOR serialisation of TimeToken encoded as a
-20-bytes string.
+: HMAC w/ SHA-224 computed over the CBOR serialisation of TimeToken encoded as
+a 28-bytes string.
```
Comment (cabo) [inline draft-birkholz-rats-epoch-markers.md:242]: Now if we could spend four more bytes to get a hash that is actually widely supported in hardware and software (SHA-256)?

File: draft-birkholz-rats-epoch-markers.md [hunk 2]
```diff
@@ -206,7 +206,13 @@ In a highly available service (e.g., a cloud attestation verifier) having to
...
 use time-synchronised servers that share a symmetric key, which produce and
 consume nonces based on coarse-grained clock ticks that are signed using the
-shared secret.
+shared secret.  This way, a nonce minted by a server in the pool can be
+processed by any other server in pool, which avoids the need for session
+"stickiness."
+
+A stateless-nonce supports the above use case by encoding a Posix time (i.e.,
+the epoch identifier), alongside a minimal set of metadata, authenticated with
+a symmetric key in a self-contained and compact (starting at 40 bytes) token.
```
Comment (cabo) [inline draft-birkholz-rats-epoch-markers.md:215]: The "starting at" is not Bourne out by the CDDL.

File: draft-birkholz-rats-epoch-markers.md (hunk 1 above)
Comment (thomas-fossati) [reply draft-birkholz-rats-epoch-markers.md:242]: I guess we could.  The whole idea behind SHA-1 was being able to do everything in 32 bytes, which fits well in "a certain RoT"'s API.

File: draft-birkholz-rats-epoch-markers.md (hunk 2 above)
Comment (thomas-fossati) [reply draft-birkholz-rats-epoch-markers.md:215]: Not sure I get your point: with a 0-length Pad (the inf), it serialises to 40 bytes.  That's what I wanted to say here.

Comment (cabo) [reply draft-birkholz-rats-epoch-markers.md:215]: I didn't see how to go beyond the 40-byte token, but I probably focused on the non-agile hash too much.

Comment (thomas-fossati) [reply draft-birkholz-rats-epoch-markers.md:215]: ok, got it, sorry, not enough coffee :-)

**Review by cabo — APPROVED** [review]
Thank you. This now looks good.
Adding optional agility might be a next step, but this one can go in.

**Pull request merged**